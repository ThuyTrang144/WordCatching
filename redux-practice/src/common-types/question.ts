import { Image } from "./image";

export interface Question {
  id: string;
  image: Image,
  answer: string;
}
