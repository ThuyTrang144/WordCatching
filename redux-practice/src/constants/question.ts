import { Question } from "@common-types/question";
import { IMAGE } from "./image";

const QUESTION: Question = {
  id: "1",
  image: IMAGE,
  answer: "Ba Hoa",
};

const QUESTION_LIST: Question[] = [
  {
    id: "1",
    image: IMAGE,
    answer: "Ba Hoa",
  },
  {
    id: "2",
    image: IMAGE,
    answer: "Ba Hoa",
  },
  {
    id: "3",
    image: IMAGE,
    answer: "Ba Hoa",
  },
  {
    id: "4",
    image: IMAGE,
    answer: "Ba Hoa",
  },
  {
    id: "5",
    image: IMAGE,
    answer: "Ba Hoa",
  },
];
export { QUESTION, QUESTION_LIST };
