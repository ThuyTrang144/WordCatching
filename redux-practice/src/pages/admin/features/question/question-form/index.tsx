import { v4 as uuidv4 } from "uuid";
import ImageUploading, { ImageListType } from "react-images-uploading";
import {
  Button,
  FormControl,
  FormErrorMessage,
  FormLabel,
  Input,
  Image,
  IconButton,
  useToast,
} from "@chakra-ui/react";
import { Question } from "@common-types/question";
import { faTrash } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { Formik, Form, useFormik } from "formik";
import { memo, useState, ChangeEvent } from "react";
import FormHeader from "./form-header";
import "./styles.css";

interface QuestionFormProps {
  title: string;
  formData?: Question;
  addNewQuestion?: any;
}

function QuestionForm({ title, formData, addNewQuestion }: QuestionFormProps) {
  const [images, setImage] = useState<ImageListType>([]);
  const [answer, setAnswer] = useState(formData?.answer);
  const toast = useToast();

  const onChangeImage = (imageList: ImageListType) => {
    setImage(imageList);
  };

  const onChangeAnswer = (event: ChangeEvent<HTMLInputElement>) => {
    setAnswer(event.target.value);
  };

  const handleAddQuestionSuccess = () => {
    toast({
      title: "Create new question successfully",
      status: "success",
      isClosable: true,
      position: "top",
    });

    setAnswer("");
    setImage([]);
  };

  const handleSubmitFormData = async () => {
    if (addNewQuestion) {
      const newQuestion = {
        id: uuidv4(),
        answer,
        image: {
          src: images[0].dataURL,
        },
      };
      await addNewQuestion(newQuestion, handleAddQuestionSuccess);
    }
  };

  return (
    <Formik
      initialValues={{ answer: "" }}
      onSubmit={handleSubmitFormData}
    >
      {({ isSubmitting }) => (
        <Form className="add-form__container">
          <FormHeader title={title} isLoading={isSubmitting} />
          <div className="add-form__body">
            <FormControl className="add-form__upload-container">
              <ImageUploading onChange={onChangeImage} value={images}>
                {({
                  imageList,
                  onImageUpload,
                  onImageRemove,
                  isDragging,
                  dragProps,
                }) => (imageList.length ? (
                  imageList.map((image, index) => (
                    <div className="add-form__upload-image" key={image.id}>
                      <Image src={image.dataURL} alt={image.alt} />
                      <IconButton
                        colorScheme="red"
                        aria-label="Search database"
                        icon={<FontAwesomeIcon icon={faTrash} />}
                        onClick={() => onImageRemove(index)}
                      />
                    </div>
                  ))
                ) : (
                  <Button
                    className="add-form__upload"
                    color={isDragging ? "red" : undefined}
                    onClick={onImageUpload}
                    {...dragProps}
                  >
                    Click to add an image or drag and drop one in this area
                  </Button>
                ))}
              </ImageUploading>
            </FormControl>
            <FormControl isRequired>
              <FormLabel htmlFor="answer">Answer</FormLabel>
              <Input
                id="answer"
                placeholder="Enter answer"
                onChange={onChangeAnswer}
                value={answer}
              />
              <FormErrorMessage>{}</FormErrorMessage>
            </FormControl>
          </div>
        </Form>
      )}
    </Formik>
  );
}

export default memo(QuestionForm);
