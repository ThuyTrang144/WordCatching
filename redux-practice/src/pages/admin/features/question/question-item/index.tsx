import {
  Checkbox, Td, Tr, Image, IconButton,
} from "@chakra-ui/react";
import { Question } from "@common-types/question";
import { faPencil, faTrash } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import isEqual from "lodash.isequal";
import { memo } from "react";

interface QuestionItemProps {
  index: number;
  question: Question;
  deleteQuestion?: (id: string) => void;
}

const QuestionItem = memo(({ question, index, deleteQuestion }: QuestionItemProps) => (
  <Tr className="question-item" data-testid="question-item">
    <Td>
      <Checkbox colorScheme="red" />
    </Td>
    <Td>{index}</Td>
    <Td>
      <Image
        borderRadius="full"
        boxSize="30px"
        src={question.image.src}
        alt={question.image.alt}
      />
    </Td>
    <Td>{question.answer}</Td>
    <Td>
      <IconButton
        variant="ghost"
        aria-label="Edit question"
        size="lg"
        icon={<FontAwesomeIcon icon={faPencil} />}
      />
      <IconButton
        aria-label="Delete question"
        size="lg"
        variant="ghost"
        icon={<FontAwesomeIcon icon={faTrash} />}
        onClick={() => deleteQuestion?.(question.id)}
      />
    </Td>
  </Tr>
), (prevProps, nextProps) => isEqual(prevProps.question, nextProps.question));

export default QuestionItem;
