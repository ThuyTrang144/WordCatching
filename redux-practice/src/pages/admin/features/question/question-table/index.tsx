import {
  Table,
  Thead,
  Tbody,
  Tr,
  Th,
  TableContainer,
  Checkbox,
} from "@chakra-ui/react";
import { Question } from "@common-types/question";
import { memo } from "react";
import isEqual from "lodash.isequal";
import QuestionItem from "../question-item";

interface QuestionTableProps {
  questions: Question[]
}

const QuestionTable = memo(({ questions }: QuestionTableProps) => {
  const renderQuestionList = (list: Question[]) => (
    list.map((item) => <QuestionItem question={item} key={item.id} />)
  );

  return (
    <TableContainer data-testid="question-table">
      <Table variant="simple">
        <Thead>
          <Tr>
            <Th><Checkbox colorScheme="red" /></Th>
            <Th>Id</Th>
            <Th>Image</Th>
            <Th>Answer</Th>
            <Th />
          </Tr>
        </Thead>
        <Tbody>
          {renderQuestionList(questions)}
        </Tbody>
      </Table>
    </TableContainer>
  );
}, (prevProps, nextProps) => isEqual(prevProps.questions, nextProps.questions));

export default QuestionTable;
