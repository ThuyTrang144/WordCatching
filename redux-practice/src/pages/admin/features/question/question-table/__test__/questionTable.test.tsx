import { QUESTION_LIST } from "@constants/question";
import { render } from "@testing-library/react";
import QuestionTable from "..";

describe("Question Table component", () => {
  test("should render Question Table component", () => {
    const { getByTestId } = render(<QuestionTable questions={QUESTION_LIST} />);
    expect(getByTestId("question-table")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<QuestionTable questions={QUESTION_LIST} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
