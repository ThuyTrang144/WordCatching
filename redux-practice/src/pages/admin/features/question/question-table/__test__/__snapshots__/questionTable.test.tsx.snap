// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Table component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="chakra-table__container css-zipzvv"
    data-testid="question-table"
  >
    <table
      class="chakra-table css-0"
      role="table"
    >
      <thead
        class="css-0"
      >
        <tr
          class="css-0"
          role="row"
        >
          <th
            class="css-0"
          >
            <label
              class="chakra-checkbox css-1elxtiq"
            >
              <input
                class="chakra-checkbox__input"
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                type="checkbox"
                value=""
              />
              <span
                aria-hidden="true"
                class="chakra-checkbox__control css-dnty2r"
              />
            </label>
          </th>
          <th
            class="css-0"
          >
            Id
          </th>
          <th
            class="css-0"
          >
            Image
          </th>
          <th
            class="css-0"
          >
            Answer
          </th>
          <th
            class="css-0"
          />
        </tr>
      </thead>
      <tbody
        class="css-0"
      >
        <tr
          class="question-item css-0"
          data-testid="question-item"
          role="row"
        >
          <td
            class="css-0"
            role="gridcell"
          >
            <label
              class="chakra-checkbox css-1elxtiq"
            >
              <input
                class="chakra-checkbox__input"
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                type="checkbox"
                value=""
              />
              <span
                aria-hidden="true"
                class="chakra-checkbox__control css-dnty2r"
              />
            </label>
          </td>
          <td
            class="css-0"
            role="gridcell"
          >
            1
          </td>
          <td
            class="css-0"
            role="gridcell"
          >
            <img
              alt="bao cao"
              class="chakra-image css-1mc5w02"
              src=""
            />
          </td>
          <td
            class="css-0"
            role="gridcell"
          >
            Ba Hoa
          </td>
          <td
            class="css-0"
            role="gridcell"
          >
            <button
              aria-label="Edit question"
              class="chakra-button css-p7mkev"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-pencil "
                data-icon="pencil"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M421.7 220.3L188.5 453.4L154.6 419.5L158.1 416H112C103.2 416 96 408.8 96 400V353.9L92.51 357.4C87.78 362.2 84.31 368 82.42 374.4L59.44 452.6L137.6 429.6C143.1 427.7 149.8 424.2 154.6 419.5L188.5 453.4C178.1 463.8 165.2 471.5 151.1 475.6L30.77 511C22.35 513.5 13.24 511.2 7.03 504.1C.8198 498.8-1.502 489.7 .976 481.2L36.37 360.9C40.53 346.8 48.16 333.9 58.57 323.5L291.7 90.34L421.7 220.3zM492.7 58.75C517.7 83.74 517.7 124.3 492.7 149.3L444.3 197.7L314.3 67.72L362.7 19.32C387.7-5.678 428.3-5.678 453.3 19.32L492.7 58.75z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <button
              aria-label="Delete question"
              class="chakra-button css-p7mkev"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-trash "
                data-icon="trash"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M135.2 17.69C140.6 6.848 151.7 0 163.8 0H284.2C296.3 0 307.4 6.848 312.8 17.69L320 32H416C433.7 32 448 46.33 448 64C448 81.67 433.7 96 416 96H32C14.33 96 0 81.67 0 64C0 46.33 14.33 32 32 32H128L135.2 17.69zM394.8 466.1C393.2 492.3 372.3 512 346.9 512H101.1C75.75 512 54.77 492.3 53.19 466.1L31.1 128H416L394.8 466.1z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </td>
        </tr>
        <tr
          class="question-item css-0"
          data-testid="question-item"
          role="row"
        >
          <td
            class="css-0"
            role="gridcell"
          >
            <label
              class="chakra-checkbox css-1elxtiq"
            >
              <input
                class="chakra-checkbox__input"
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                type="checkbox"
                value=""
              />
              <span
                aria-hidden="true"
                class="chakra-checkbox__control css-dnty2r"
              />
            </label>
          </td>
          <td
            class="css-0"
            role="gridcell"
          >
            2
          </td>
          <td
            class="css-0"
            role="gridcell"
          >
            <img
              alt="bao cao"
              class="chakra-image css-1mc5w02"
              src=""
            />
          </td>
          <td
            class="css-0"
            role="gridcell"
          >
            Ba Hoa
          </td>
          <td
            class="css-0"
            role="gridcell"
          >
            <button
              aria-label="Edit question"
              class="chakra-button css-p7mkev"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-pencil "
                data-icon="pencil"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M421.7 220.3L188.5 453.4L154.6 419.5L158.1 416H112C103.2 416 96 408.8 96 400V353.9L92.51 357.4C87.78 362.2 84.31 368 82.42 374.4L59.44 452.6L137.6 429.6C143.1 427.7 149.8 424.2 154.6 419.5L188.5 453.4C178.1 463.8 165.2 471.5 151.1 475.6L30.77 511C22.35 513.5 13.24 511.2 7.03 504.1C.8198 498.8-1.502 489.7 .976 481.2L36.37 360.9C40.53 346.8 48.16 333.9 58.57 323.5L291.7 90.34L421.7 220.3zM492.7 58.75C517.7 83.74 517.7 124.3 492.7 149.3L444.3 197.7L314.3 67.72L362.7 19.32C387.7-5.678 428.3-5.678 453.3 19.32L492.7 58.75z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <button
              aria-label="Delete question"
              class="chakra-button css-p7mkev"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-trash "
                data-icon="trash"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M135.2 17.69C140.6 6.848 151.7 0 163.8 0H284.2C296.3 0 307.4 6.848 312.8 17.69L320 32H416C433.7 32 448 46.33 448 64C448 81.67 433.7 96 416 96H32C14.33 96 0 81.67 0 64C0 46.33 14.33 32 32 32H128L135.2 17.69zM394.8 466.1C393.2 492.3 372.3 512 346.9 512H101.1C75.75 512 54.77 492.3 53.19 466.1L31.1 128H416L394.8 466.1z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </td>
        </tr>
        <tr
          class="question-item css-0"
          data-testid="question-item"
          role="row"
        >
          <td
            class="css-0"
            role="gridcell"
          >
            <label
              class="chakra-checkbox css-1elxtiq"
            >
              <input
                class="chakra-checkbox__input"
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                type="checkbox"
                value=""
              />
              <span
                aria-hidden="true"
                class="chakra-checkbox__control css-dnty2r"
              />
            </label>
          </td>
          <td
            class="css-0"
            role="gridcell"
          >
            3
          </td>
          <td
            class="css-0"
            role="gridcell"
          >
            <img
              alt="bao cao"
              class="chakra-image css-1mc5w02"
              src=""
            />
          </td>
          <td
            class="css-0"
            role="gridcell"
          >
            Ba Hoa
          </td>
          <td
            class="css-0"
            role="gridcell"
          >
            <button
              aria-label="Edit question"
              class="chakra-button css-p7mkev"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-pencil "
                data-icon="pencil"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M421.7 220.3L188.5 453.4L154.6 419.5L158.1 416H112C103.2 416 96 408.8 96 400V353.9L92.51 357.4C87.78 362.2 84.31 368 82.42 374.4L59.44 452.6L137.6 429.6C143.1 427.7 149.8 424.2 154.6 419.5L188.5 453.4C178.1 463.8 165.2 471.5 151.1 475.6L30.77 511C22.35 513.5 13.24 511.2 7.03 504.1C.8198 498.8-1.502 489.7 .976 481.2L36.37 360.9C40.53 346.8 48.16 333.9 58.57 323.5L291.7 90.34L421.7 220.3zM492.7 58.75C517.7 83.74 517.7 124.3 492.7 149.3L444.3 197.7L314.3 67.72L362.7 19.32C387.7-5.678 428.3-5.678 453.3 19.32L492.7 58.75z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <button
              aria-label="Delete question"
              class="chakra-button css-p7mkev"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-trash "
                data-icon="trash"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M135.2 17.69C140.6 6.848 151.7 0 163.8 0H284.2C296.3 0 307.4 6.848 312.8 17.69L320 32H416C433.7 32 448 46.33 448 64C448 81.67 433.7 96 416 96H32C14.33 96 0 81.67 0 64C0 46.33 14.33 32 32 32H128L135.2 17.69zM394.8 466.1C393.2 492.3 372.3 512 346.9 512H101.1C75.75 512 54.77 492.3 53.19 466.1L31.1 128H416L394.8 466.1z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </td>
        </tr>
        <tr
          class="question-item css-0"
          data-testid="question-item"
          role="row"
        >
          <td
            class="css-0"
            role="gridcell"
          >
            <label
              class="chakra-checkbox css-1elxtiq"
            >
              <input
                class="chakra-checkbox__input"
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                type="checkbox"
                value=""
              />
              <span
                aria-hidden="true"
                class="chakra-checkbox__control css-dnty2r"
              />
            </label>
          </td>
          <td
            class="css-0"
            role="gridcell"
          >
            4
          </td>
          <td
            class="css-0"
            role="gridcell"
          >
            <img
              alt="bao cao"
              class="chakra-image css-1mc5w02"
              src=""
            />
          </td>
          <td
            class="css-0"
            role="gridcell"
          >
            Ba Hoa
          </td>
          <td
            class="css-0"
            role="gridcell"
          >
            <button
              aria-label="Edit question"
              class="chakra-button css-p7mkev"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-pencil "
                data-icon="pencil"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M421.7 220.3L188.5 453.4L154.6 419.5L158.1 416H112C103.2 416 96 408.8 96 400V353.9L92.51 357.4C87.78 362.2 84.31 368 82.42 374.4L59.44 452.6L137.6 429.6C143.1 427.7 149.8 424.2 154.6 419.5L188.5 453.4C178.1 463.8 165.2 471.5 151.1 475.6L30.77 511C22.35 513.5 13.24 511.2 7.03 504.1C.8198 498.8-1.502 489.7 .976 481.2L36.37 360.9C40.53 346.8 48.16 333.9 58.57 323.5L291.7 90.34L421.7 220.3zM492.7 58.75C517.7 83.74 517.7 124.3 492.7 149.3L444.3 197.7L314.3 67.72L362.7 19.32C387.7-5.678 428.3-5.678 453.3 19.32L492.7 58.75z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <button
              aria-label="Delete question"
              class="chakra-button css-p7mkev"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-trash "
                data-icon="trash"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M135.2 17.69C140.6 6.848 151.7 0 163.8 0H284.2C296.3 0 307.4 6.848 312.8 17.69L320 32H416C433.7 32 448 46.33 448 64C448 81.67 433.7 96 416 96H32C14.33 96 0 81.67 0 64C0 46.33 14.33 32 32 32H128L135.2 17.69zM394.8 466.1C393.2 492.3 372.3 512 346.9 512H101.1C75.75 512 54.77 492.3 53.19 466.1L31.1 128H416L394.8 466.1z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </td>
        </tr>
        <tr
          class="question-item css-0"
          data-testid="question-item"
          role="row"
        >
          <td
            class="css-0"
            role="gridcell"
          >
            <label
              class="chakra-checkbox css-1elxtiq"
            >
              <input
                class="chakra-checkbox__input"
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                type="checkbox"
                value=""
              />
              <span
                aria-hidden="true"
                class="chakra-checkbox__control css-dnty2r"
              />
            </label>
          </td>
          <td
            class="css-0"
            role="gridcell"
          >
            5
          </td>
          <td
            class="css-0"
            role="gridcell"
          >
            <img
              alt="bao cao"
              class="chakra-image css-1mc5w02"
              src=""
            />
          </td>
          <td
            class="css-0"
            role="gridcell"
          >
            Ba Hoa
          </td>
          <td
            class="css-0"
            role="gridcell"
          >
            <button
              aria-label="Edit question"
              class="chakra-button css-p7mkev"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-pencil "
                data-icon="pencil"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M421.7 220.3L188.5 453.4L154.6 419.5L158.1 416H112C103.2 416 96 408.8 96 400V353.9L92.51 357.4C87.78 362.2 84.31 368 82.42 374.4L59.44 452.6L137.6 429.6C143.1 427.7 149.8 424.2 154.6 419.5L188.5 453.4C178.1 463.8 165.2 471.5 151.1 475.6L30.77 511C22.35 513.5 13.24 511.2 7.03 504.1C.8198 498.8-1.502 489.7 .976 481.2L36.37 360.9C40.53 346.8 48.16 333.9 58.57 323.5L291.7 90.34L421.7 220.3zM492.7 58.75C517.7 83.74 517.7 124.3 492.7 149.3L444.3 197.7L314.3 67.72L362.7 19.32C387.7-5.678 428.3-5.678 453.3 19.32L492.7 58.75z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <button
              aria-label="Delete question"
              class="chakra-button css-p7mkev"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-trash "
                data-icon="trash"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M135.2 17.69C140.6 6.848 151.7 0 163.8 0H284.2C296.3 0 307.4 6.848 312.8 17.69L320 32H416C433.7 32 448 46.33 448 64C448 81.67 433.7 96 416 96H32C14.33 96 0 81.67 0 64C0 46.33 14.33 32 32 32H128L135.2 17.69zM394.8 466.1C393.2 492.3 372.3 512 346.9 512H101.1C75.75 512 54.77 492.3 53.19 466.1L31.1 128H416L394.8 466.1z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
